packages/devextreme/testing/tests/DevExpress.ui.widgets/tabs.tests.js

```diff
@@ -422,6 +423,200 @@ QUnit.module('General', () => {
 });
 
 QUnit.module('Tab select action', () => {
+    QUnit.module('onSelectionChanging', {
+        beforeEach() {
+            this.onSelectionChangingSpy = sinon.stub();
+            this.onSelectionChangedSpy = sinon.stub();
+            const initialOptions = {
+                items: [
+                    { text: '0' },
+                    { text: '1' },
+                    { text: '2' },
+                    { text: '3' }
+                ],
+                onSelectionChanging: this.onSelectionChangingSpy,
+                onSelectionChanged: this.onSelectionChangedSpy,
+                selectByClick: true,
+                selectedIndex: 0
+            };
+            this.init = (options) => {
+                this.$tabs = $('#tabs');
+                this.tabInstance = this.$tabs
+                    .dxTabs($.extend({}, initialOptions, options))
+                    .dxTabs('instance');
+            };
+            this.reinit = (options) => {
+                this.tabInstance.dispose();
+
+                this.init(options);
+            };
+
+            this.init();
+            this.clock = sinon.useFakeTimers();
+            this.CANCEL_DELAY = 0;
+        },
+        afterEach() {
+            this.clock.restore();
+        }
+    }, () => {
+        QUnit.test('Selection should remain unchanged when onSelectionChanging is cancelled initially', function(assert) {
+            let count = 0;
+            const onSelectionChangingSpy = sinon.spy(function(e) {
+                e.cancel = true;
+                count++;
+            });
+            const onSelectionChangedSpy = sinon.spy();
+            this.reinit({
+                onSelectionChanging: onSelectionChangingSpy,
+                onSelectionChanged: onSelectionChangedSpy,
+            });
+
+            const $item = this.$tabs.find(`.${TABS_ITEM_CLASS}`).eq(1);
+            $item.trigger('dxclick');
+
+            assert.strictEqual(count, 2, 'onSelectionChanging should be called twice');
```

Comment:
[incorrect logic]
I saw your comment about this question and I understand that Selection's `onSelectionChanged` is fired on initialized. However, I see that Collection's `onSelectionChanged` is not fired, and I don't see any product reason to fire it on init because selection is anyway applied. So I ask you to stop firing selectionChanging on init.